### PR TITLE
Postprocessing update: add output variables related to peak intensity & savings and emission intensity & savings

### PR DESCRIPTION
**Pull request by JieXiong9119**
Pull request overview
---------------------

This PR is to implement and register multiple output variables related to peak load and emission data, regarding calculated intensity and savings (and percent savings), and write to comstockwide csv in the postprocessing workflow. Changes take place in 
- calculation in [postprocessing/comstockpostproc/comstock.py](https://github.com/NREL/ComStock/blob/4d7baa7d1009afa3689ed0c120c4ae8b9f1a1719/postprocessing/comstockpostproc/comstock.py)
- naming constant conversion in [postprocessing/comstockpostproc/naming_mixin.py](https://github.com/NREL/ComStock/blob/4d7baa7d1009afa3689ed0c120c4ae8b9f1a1719/postprocessing/comstockpostproc/naming_mixin.py)
- unit conversion in [postprocessing/comstockpostproc/units_mixin.py](https://github.com/NREL/ComStock/blob/4d7baa7d1009afa3689ed0c120c4ae8b9f1a1719/postprocessing/comstockpostproc/units_mixin.py)
- register entries to be written in final output (comstock wide.csv) in [postprocessing/comstockpostproc/resources/comstock_column_definitions.csv](https://github.com/NREL/ComStock/blob/4d7baa7d1009afa3689ed0c120c4ae8b9f1a1719/postprocessing/comstockpostproc/resources/comstock_column_definitions.csv)

Changes I did not make that might be relavent:
- weighted savings calculation for peak load and emissions
- previously existing function [`add_normalized_qoi_columns`](https://github.com/NREL/ComStock/blob/4d7baa7d1009afa3689ed0c120c4ae8b9f1a1719/postprocessing/comstockpostproc/comstock.py#L1708) calculating for a few peak load variables remains untouched, due to unit scale difference (w/ft2 in existing function, but kw/ft2 in added function)

### Pull Request Author

This pull request makes changes to (select all the apply):
 - [ ] Documentation
 - [ ] Infrastructure (includes apptainer image, buildstock batch, dependencies, continuous integration tests)
 - [ ] Sampling
 - [ ] Workflow Measures
 - [ ] Upgrade Measures
 - [ ] Reporting Measures
 - [x] Postprocessing

Author pull request checklist:

 - [ ] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [ ] Added tests for new measures
 - [ ] Updated measure .xml(s)
 - [x] Register values added to `comstock_column_definitions.csv`
 - [ ] Both `options_lookup.tsv` files updated
 - [ ] 10k+ test run
 - [ ] Change documentation written
 - [ ] Measure documentation written
 - [ ] ComStock documentation updated
 - [ ] Changes reflected in example `.yml` files
 - [ ] Changes reflected in `README.md` files
 - [ ] Added 'See ComStock License' language to first two lines of each code file
 - [ ] Implements corresponding measure tests and indexing path in `test/reporting_measure_tests.txt`, `test/workflow_measure_tests.txt`, or `test/upgrade_measure_tests.txt`
 - [ ] All new and existing tests pass the CI

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a code review on GitHub
 - [ ] All related changes have been implemented: data and method additions, changes, tests
 - [ ] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] Reviewed change documentation
 - [ ] Ensured code files contain License reference
 - [ ] Results differences are reasonable
 - [ ] Make sure the newly added measures has been added with tests and indexed properly
 - [ ] CI status: all tests pass

#### ComStock Licensing Language - Add to Beginning of Each Code File
```
# ComStock™, Copyright (c) 2023 Alliance for Sustainable Energy, LLC. All rights reserved.
# See top level LICENSE.txt file for license terms.
```

### CODE REVIEW

**Comment by asparke2** [comment]
@JieXiong9119 for processing the 2024 Release 2 SDR dataset we are making changes on this branch: https://github.com/NREL/ComStock/tree/wenyi/euss_r3, so please change the target branch of this PR to point to that instead of main. I already added the 15 `daily_average_electricity_ghg_emissions` columns back to that branch. Also, adding 215 calculated columns to all the public files is too many because these variables are not likely to be used by most people, so we can keep them in the column_definitions.csv and keep the code which creates them, but change the export options to FALSE, FALSE so that they don't end up in the public dataset. If you want to run postprocessing separately yourself with them set to TRUE, FALSE that's fine.

File: postprocessing/comstockpostproc/resources/comstock_column_definitions.csv [hunk 1]
```diff
@@ -1611,12 +1611,227 @@ calculated,out.utility_bills.electricity_bill_intensity,out.utility_bills.electr
...
 calculated,calc.percent_savings.electricity.utility_bills.electricity_bill_mean_intensity,calc.percent_savings.electricity.utility_bills.electricity_bill_mean_intensity,TRUE,FALSE,float,percent,percent,Mean electric bill per floor area
 calculated,calc.percent_savings.electricity.utility_bills.electricity_bill_min_intensity,calc.percent_savings.electricity.utility_bills.electricity_bill_min_intensity,TRUE,FALSE,float,percent,percent,Minimum electric bill per floor area
 calculated,calc.percent_savings.electricity.utility_bills.electricity_bill_max_intensity,calc.percent_savings.electricity.utility_bills.electricity_bill_max_intensity,TRUE,FALSE,float,percent,percent,Max electric bill per floor area
-calculated,out.utility_bills.electricity_bill_max_intensity..usd_per_ft2,out.utility_bills.electricity_bill_max_intensity..usd_per_ft2,TRUE,FALSE,float,percent,percent,Max total electricity bill intensity across all electricity rates in US dollars per square foot
```
Comment (JieXiong9119) [inline postprocessing/comstockpostproc/resources/comstock_column_definitions.csv:1618]: I believe this is a duplicated row (with extra string) with line 1835? @asparke2

File: postprocessing/comstockpostproc/resources/comstock_column_definitions.csv [hunk 2]
```diff
@@ -1611,12 +1611,227 @@ calculated,out.utility_bills.electricity_bill_intensity,out.utility_bills.electr
...
 calculated,calc.utility_bills.total_mean_bill,calc.utility_bills.total_mean_bill,TRUE,FALSE,float,usd,usd,Total utility bill across all fuel types using mean electricity costs across all electricity utility rates for the model.
+calculated,calc.utility_bills.total_mean_bill_savings,calc.utility_bills.total_mean_bill_savings,TRUE,FALSE,float,usd,usd,Savings of total utility bill across all fuel types using mean electricity costs across all electricity utility rates for the model.
```
Comment (JieXiong9119) [inline postprocessing/comstockpostproc/resources/comstock_column_definitions.csv:1834]: I would like to add this row for general output, but feel free to have different idea and make change to it @asparke2.

File: postprocessing/comstockpostproc/resources/comstock_column_definitions.csv [hunk 3]
```diff
@@ -1628,6 +1843,9 @@ calculated,out.utility_bills.electricity_bill_savings_median,out.utility_bills.e
 calculated,out.utility_bills.electricity_bill_savings_median_intensity,out.utility_bills.electricity_bill_savings_median_intensity,FALSE,FALSE,float,usd_per_ft2,usd_per_ft2,Electricity bill savings intensity in 2022 usd/sqft using the median applicable electricity rate.
 calculated,out.utility_bills.electricity_bill_savings_min,out.utility_bills.electricity_bill_savings_min,FALSE,FALSE,float,usd,usd,Electricity bill savings in 2022 usd using the minimum applicable electricity rate.
 calculated,out.utility_bills.electricity_bill_savings_min_intensity,out.utility_bills.electricity_bill_savings_min_intensity,FALSE,FALSE,float,usd_per_ft2,usd_per_ft2,Electricity bill savings intensity in 2022 usd/sqft using the minimum applicable electricity rate.
+calculated,out.utility_bills.fuel_oil_bill_savings,out.utility_bills.fuel_oil_bill_savings,TRUE,FALSE,float,usd,usd,Total fuel oil bill savings
+calculated,out.utility_bills.natural_gas_bill_savings,out.utility_bills.natural_gas_bill_savings,TRUE,FALSE,float,usd,usd,Total natural gas bill savings
+calculated,out.utility_bills.propane_bill_savings,out.utility_bills.propane_bill_savings,TRUE,FALSE,float,usd,usd,Total propane bill savings
```
Comment (JieXiong9119) [inline postprocessing/comstockpostproc/resources/comstock_column_definitions.csv:1848]: These 3 rows are tricky. Feel free to disable them @asparke2.

**Review by asparke2 — APPROVED** [review]

**Pull request merged**